src/main/java/org/openrewrite/kotlin/internal/KotlinTreeParserVisitor.java

```diff
@@ -1644,7 +1647,22 @@ public J visitKtFile(KtFile file, ExecutionContext data) {
         List<JRightPadded<Statement>> statements = new ArrayList<>(file.getDeclarations().size());
         List<KtDeclaration> declarations = file.getDeclarations();
         for (KtDeclaration declaration : declarations) {
-            Statement statement = convertToStatement(declaration.accept(this, data));
+            Statement statement;
+            try {
+                statement = convertToStatement(declaration.accept(this, data));
+            } catch (Exception e) {
+                statement = new J.Unknown(
+                        randomId(),
+                        deepPrefix(declaration),
```

Comment:
The prefix of the declaration is retrieved through the prefix method.
Then a substring of the file text is created using PsiUtilsKt.getStartOffsetSkippingComments(declaration) and the declarations endoffset.